src/Catalog/Telemetry/TelemetryService.cs

```diff
@@ -20,44 +23,14 @@ public class TelemetryService : ITelemetryService
         private const string StatusCode = "StatusCode";
         private const string ContentLength = "ContentLength";
 
-        private const string CatalogIndexReadDurationSeconds = "CatalogIndexReadDurationSeconds";
-
-        private const string CatalogIndexWriteDurationSeconds = "CatalogIndexWriteDurationSeconds";
+        public IDictionary<string, string> GlobalDimensions { get; private set; }
 
         public TelemetryService(TelemetryClient telemetryClient)
         {
             _telemetryClient = telemetryClient ?? throw new ArgumentNullException(nameof(telemetryClient));
-        }
-
-        public void TrackHttpHeaderDuration(
-            TimeSpan duration,
-            HttpMethod method,
-            Uri uri,
-            bool success,
-            HttpStatusCode? statusCode,
-            long? contentLength)
-        {
-            if (method == null)
-            {
-                throw new ArgumentNullException(nameof(method));
-            }
-
-            if (uri == null)
-            {
-                throw new ArgumentNullException(nameof(uri));
-            }
+            _telemetryClientWrapper = new TelemetryClientWrapper(_telemetryClient);
 
-            _telemetryClient.TrackMetric(
-                HttpHeaderDurationSeconds,
-                duration.TotalSeconds,
-                new Dictionary<string, string>
-                {
-                    { Method, method.ToString() },
-                    { Uri, uri.AbsoluteUri },
-                    { StatusCode, ((int?)statusCode)?.ToString() },
-                    { Success, success.ToString() },
-                    { ContentLength, contentLength?.ToString() }
-                });
+            GlobalDimensions = new Dictionary<string, string>();
         }
 
         public void TrackCatalogIndexReadDuration(TimeSpan duration, Uri uri)
```

Comment:
Given more time I would replace the usage of TrackCatalogIndexReadDuration, TrackCatalogIndexWriteDuration with TrackDuration so we won't have to initilize Stopwatch in the code.